### PR TITLE
PP-363: Decisions search filter fixes

### PR DESCRIPTION
**Pull request by juho-lehmonen**
**Drupal setup**
- On your local drupal installation, checkout this branch: `PP-363-search-filter-fixes`
- Run `make drush-cim drush-cr`
- If you don't have any data, run `make ahjo-migrations` (NOTE: you'll need the local proxy URL for this one. Also, edit the user permissions on the test site and add the following permissions for anonymous users: "Access external documents from Ahjo API")
- Rebuild and reindex the decisions index: https://helsinki-paatokset.docker.so/fi/admin/config/search/search-api/index/decisions

**To test**
- Checkout branch and start the app with `npm start` (make sure the index is set for decisions)
- Read the code changes. This has a lot of changes to the category and DM selection, so those parts of the code should especially be checked

**Date as selected filter**
- Test the date selector. The selected values should appear as a tag in the selected filters area
- You should be able to reset the date search by deselecting the tag

**All decisionmakers filter, deselecting decisionmakers**
- There should be a new item in the decisionmakers list: "Kaikki päättäjät"
  - Selecting this acts the same as having the filter empty, but should appear as a tag and shouldnt cause any errors
- You should now be able to deselect the selected item in the decisionmaker dropdown by clicking it again

**ID based decisionmaker and category selection**
- Decisionmaker and top category selection should now be based on the indexed "Sector ID" and "Top category code" fields
- Compare the fi and sv search results to the version on the test site:
  - https://nginx-paatokset-test.agw.arodevtest.hel.fi/fi/paattajat
  - https://nginx-paatokset-test.agw.arodevtest.hel.fi/sv/beslutsfattare
- On the test site, you should get vastly different results when filtering by category and decision maker (not the special case dm selections)
- On the local search app, the results should be more similar between language versions (they might vary by a few decisions, though)
- Also test that the URL query parameters work by reloading the page after adding filters

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+REACT_APP_NAME=paatokset_decisions
+REACT_APP_ELASTIC_URL=http://localhost:9200
```
Comment (jeremysteerio) [inline .env]: Let's not commit .env 😁

**Comment by jeremysteerio** [comment]
Looks good to me! Couple of notes that I came across:

- When de-selecting predefined dates (last week, last month etc.) by pressing the pill, the selection remains in the dropdown.
- Drupal dropdown styles seems to be having some side effects on the policymakers dropdown: https://nginx-paatokset-test.agw.arodevtest.hel.fi/fi/paattajat (hovering selection causes both bg and text colors to be black)

**Comment by jeremysteerio** [comment]
Looks good to me! Couple of notes that I came across:

- When de-selecting predefined dates (last week, last month etc.) by pressing the pill, the selection remains in the dropdown.
- Drupal dropdown styles seems to be having some side effects on the policymakers dropdown: https://nginx-paatokset-test.agw.arodevtest.hel.fi/fi/paattajat (hovering selection causes both bg and text colors to be black)

**Review by jeremysteerio — CHANGES_REQUESTED** [review]

**Comment by juho-lehmonen** [comment]
>     * When de-selecting predefined dates (last week, last month etc.) by pressing the pill, the selection remains in the dropdown.

This should be fixed now. Can you verify that it works for you too?

>     * Drupal dropdown styles seems to be having some side effects on the policymakers dropdown: https://nginx-paatokset-test.agw.arodevtest.hel.fi/fi/paattajat (hovering selection causes both bg and text colors to be black)

Those should be fixed on the local search fixes branch.

**Pull request merged**